pkg/curatedpackages/reader.go

```diff
@@ -122,13 +151,29 @@ func (r *PackageReader) fetchImagesFromBundle(ctx context.Context, versionsBundl
 				OS:          ctrl.OS,
 				OSName:      ctrl.OSName,
 				URI:         fmt.Sprintf("%s/%s@%s", getRegistry(ctrl.URI), version.Repository, version.Digest),
+				ImageDigest: version.Digest,
 			}
 			images = append(images, image)
 		}
 	}
 	return images, nil
 }
 
+func removeDuplicateImages(strSlice []releasev1.Image) []releasev1.Image {
+	allKeys := make(map[string]releasev1.Image)
```

Comment:
Nit: there's no need to copy the images into the map. You could just use:
```go
map[string]struct{}
```
Which wouldn't have to copy as many things around, saving both memory and time.
That would change line 167 to something more like: 
```go
allKeys[item.VersionedImage()] = struct{}{}
```

Reply:
Done.